app/models/lettings_log.rb

```diff
@@ -40,6 +40,13 @@ class LettingsLog < Log
   belongs_to :managing_organisation, class_name: "Organisation", optional: true
 
   scope :filter_by_year, ->(year) { where(startdate: Time.zone.local(year.to_i, 4, 1)...Time.zone.local(year.to_i + 1, 4, 1)) }
+  scope :filter_by_years_or_nil, lambda { |years, _user = nil|
```

Comment:
can we put this on the log model instead?

Reply:
(given we can use startdate = saledate for sales logs)

Reply:
I'm not sure I fully understand, it wouldn't work for sales if I move this method as it is right now, cause startdate is not a column in sales logs. Is there a change to it that you had in mind that would make it work for both?